lib/surface/components/form/error_tag.ex

```diff
@@ -124,7 +124,25 @@ defmodule Surface.Components.Form.ErrorTag do
     # Because the error messages we show in our forms and APIs
     # are defined inside Ecto, we need to translate them dynamically.
     Enum.reduce(opts, msg, fn {key, value}, acc ->
-      String.replace(acc, "%{#{key}}", to_string(value))
+      try do
+        String.replace(acc, "%{#{key}}", to_string(value))
+      rescue
+        e in ArgumentError ->
+          if e.message =~ "cannot convert the given list to a string" do
```

Comment:
Unfortunalty, this condition will not catch the error from #450

> ** (Protocol.UndefinedError) protocol String.Chars not implemented for {:parameterized, Ecto.Enum, %{on_dump: %{B2B: "B2B", EoC: "EoC"}, on_load: %{"B2B" => :B2B, "EoC" => :EoC}, type: :string, values: [:B2B, :EoC]}} of type Tuple

I do not like the idea to have a list of strings that will produce this error, but I don't know how we can solve this in a better way.

Reply:
Could we provide a more general error message instead, then we can append to all `ArgumentError` raised?

Reply:
The error type from #450 is not an `ArgumentError` but `Protocol.UndifinedError` :(

Reply:
What if we instead of catching the error and re-raising it, we test the value, show a fallback message and warn with the hint? 

Something like:

```elixir
    Enum.reduce(opts, msg, fn {key, value}, acc ->
      message = 
        if String.Chars.impl_for(value) do
          to_string(value)
        else
          # TODO: show_warn_with_hint(...)
          "invalid value"
        end
      String.replace(acc, "%{#{key}}", message)
    end)
```

Reply:
It won't work for the `unsafe_unique` validator as it returns a list and `List` implements the `String.Chars` protocol, so the condition will be valid and the `to_string` function will return the plain old `ArgumentError` :cry: 

https://github.com/elixir-lang/elixir/blob/7e97d08b03129f367a6578ebacb17861b6726e4f/lib/elixir/lib/list.ex#L967-L996

Reply:
@Malian in this case, we have a few options:

1. Catch any error raised (not only ArgumentError), return the generic "invalid value" message as suggested and emit a warning. Something like:

    ```
    the fallback message translator for the `ErrorTag` component cannot handle the given message.

    Hint: If you're using `GetText`, set up the default translator to your application helpers:

      config :surface, :components, [
        {Surface.Components.Form.ErrorTag, default_translator: {MyAppWeb.ErrorHelpers, :translate_error}}
      ]

    Original error: #{Exception.message(exception)
    ```

2. Do the above if an error occurs, but additionally emit another warning earlier even if we can handle the message but the default_translator is not set. This way users will knows upfront that they need to either set up `GetText` or roll their own translator, otherwise the fallback may not handle the message properly. Something like:

    ```
    the default message translator for `ErrorTag` has not been set. Using a fallback translator.

    Pay attention that the fallback translator may not be able to translate all given messages.

    Hint: If you're using `GetText`, set up the default translator to your application helpers with:

      config :surface, :components, [
        {Surface.Components.Form.ErrorTag, default_translator: {MyAppWeb.ErrorHelpers, :translate_error}}
      ]

    If you want to suppress this warning and keep using the fallback translator, please explicitly set it 
    as the default translator  with: 

      config :surface, :components, [
        {Surface.Components.Form.ErrorTag, default_translator: {Surface.Components.Form.ErrorTag, :translate_error}}
      ]
    ```

Reply:
Actually, I changed my mind and vote for `#1`. I believe if the docs already mention that the fallback translator may fail, we don't have to always warn, only when it fails.

Reply:
I confess that I also prefer the first solution. I would not like to emit a warning if the `default_translator` has not been set for the `ErrorTag` component, as some user would not use the built-in form components.

I'll update the doc and the PR according to your proposition.